### PR TITLE
Small refactors and bug fixes.

### PR DESCRIPTION
**Pull request by jroesch**
I'm starting the process of upstreaming the bits and pieces of Relay we developed while prototyping new features. This PR is composed of bug fixes and refactors which should make it easier to port those changes.

### CODE REVIEW

File: src/relay/ir/expr.cc [hunk 1]
```diff
@@ -135,6 +135,12 @@ FuncType FunctionNode::func_type_annotation() const {
   return FuncTypeNode::make(param_types, this->ret_type, this->type_params, {});
 }
 
+bool FunctionNode::IsPrimitive() const {
+   NodeRef res = FunctionGetAttr(GetRef<Function>(this), "Primitive");
+   const ir::IntImm* pval = res.as<ir::IntImm>();
+     return pval && pval->value != 0;
```
Comment (wweic) [inline src/relay/ir/expr.cc]: I think the indentation is not needed.

File: include/tvm/relay/expr.h [hunk 2]
```diff
@@ -248,6 +248,13 @@ class FunctionNode : public ExprNode {
    */
   TVM_DLL FuncType func_type_annotation() const;
 
+  /*!
+   * \brief Check whether the function is a primitive function.
+   *
+   * \return Whether the function is primitive or not.
+   */
+  bool IsPrimitive() const;
```
Comment (wweic) [inline include/tvm/relay/expr.h]: Maybe replace this line with the new API?
https://github.com/dmlc/tvm/blob/01cfdb7ebcef0abc9282f68c3c4484974a6cc6e1/src/relay/pass/fuse_ops.cc#L702

Comment (jroesch) [reply include/tvm/relay/expr.h]: Yeah I was cherry-picking changes from an old branch 😄 got distracted at TVM conference while working on this PR.

File: src/relay/ir/expr.cc (hunk 1 above)
Comment (wweic) [reply src/relay/ir/expr.cc]: also fix this since it's failing CI lint test.

**Comment by tqchen** [comment]
Thanks, @wweic @jroesch , this is now merged

**Pull request merged**